checker/manual/external-tools.tex

```diff
@@ -324,87 +336,103 @@ \section{Maven\label{maven}}
+
+\begin{alltt}
+    <plugin>
+        <!-- This plugin will set properties values using dependency information -->
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <version>2.3</version>
+        <executions>
+            <execution>
+                <goals>
+                    <goal>properties</goal>
+                </goals>
+            </execution>
+        </executions>
+    </plugin>
+\end{alltt}
 
-This script assumes that the Checker Framework is the only annotation processor being run.
-If this is not the case, please modify the \code{javac\_maven} script accordingly.
 
-\item Create a text file named \code{argfile} in the same directory as the \code{pom.xml}
-and include in it the command-line parameters to pass to the Java compiler.
+\item Direct the Maven compiler plugin to use the desired checkers.
+Change the reference to the \code{maven-compiler-plugin} within the \code{<plugins>}
+section, or add it if it is not present.
 
-Example contents of \code{argfile}:
+For example, to use the \code{org.checkerframework.checker.nullness.NullnessChecker}:
 
 \begin{alltt}
--processor org.checkerframework.checker.nullness.NullnessChecker
--AsuppressWarnings=purity.invalid.overriding
--Alint
--AprintErrorStack
--Awarns
--Xmaxwarns 10000
+    <plugin>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.3</version>
+        <configuration>
+            <!-- Change source and target to 1.7 if using Java 7 -->
+            <source>1.8</source>
+            <target>1.8</target>
+            <fork>true</fork>
+            <annotationProcessors>
+                <!-- Add all the checkers you want to enable here -->
+                <annotationProcessor>org.checkerframework.checker.nullness.NullnessChecker</annotationProcessor>
+            </annotationProcessors>
+            <compilerArgs>
+                <!-- location of the annotated JDK, which comes from a Maven dependency -->
+                <arg>-Xbootclasspath/p:$\{annotatedJdk\}</arg>
+                <arg>-J-Xbootclasspath/p:$\{jsr308Javac\}</arg>
+            </compilerArgs>
+        </configuration>
+    </plugin>
 \end{alltt}
 
-Due to limitations of the \code{pom.xml} syntax, some command-line options are
-difficult or impossible to pass via the \code{pom.xml}, hence the need for this file.
+Now, building with Maven should run the checkers during compilation.
+
+Notice that using this approach, no external setup is necessary,
+so your Maven build should be reproducible in any server.
 
-If you would like to call this file something other than \code{argfile},
-please modify the \code{javac\_maven} script accordingly.
+If you want to allow Maven to compile your code without running the checkers, you may
+want to move the declarations above to within a Maven profile, so that the checkers
+would only run if the profile was enabled.
 
 \end{enumerate}
 
 %% Of historical interest only.
```

Comment:
Given that your diff removes the text this comment is about, I think the comment should be removed too, or else people reading the LaTeX source will be confused.